opm/parser/eclipse/EclipseState/Schedule/ScheduleState.hpp

```diff
@@ -150,6 +150,15 @@ namespace Opm {
             }
 
 
+            const T* find(const std::function<bool(const std::pair<K,std::shared_ptr<T>>&)>& predicate) const {
```

Comment:
Just for the record, using this parameter type (`std::pair<K, shared_ptr<T>>`) means we'll copy all elements of `m_data` when calling the predicate function through `std::find_if()`.  The element type of an `unordered_map<K, shared_ptr<T>>` is
```C++
std::pair<const K, shared_ptr<T>>
```
which is incompatible with the declared parameter type of the `std::function`.  Since we're already in a header, this is a case where I'd recommend **not** using `std::function`, but rather to take a callable by perfect forwarding:
```C++
template <typename Predicate>
const T* find(Predicate&& predicate) const
{
    auto iter = std::find_if(this->m_data.begin(), this->m_data.end(),
                             std::forward<Predicate>(predicate));
    // Rest as before
}
```
Then the call site can use lambdas, `std::function`s or custom predicate types.

Reply:
Thank you - update here: https://github.com/OPM/opm-common/pull/2286